### PR TITLE
run snyk on github action pull_request

### PR DESCRIPTION
**Pull request by jamesgorrie**
part of #3882

## What does this change?
This runs our snyk github workflow on `pull_request` rather than on `push`.

## Why?
We would like to have the @dependabot chore PRs have all green checks. This runs @dependabot on `pull_request` instead of `push` as it then runs in `write` mode instead of `read-only` mode.

It seems a little convoluted, but I didn't want to lose running CodeQL on each push, so have opted to keep that and ignore dependabot PRs.

[The error](https://github.com/guardian/dotcom-rendering/runs/4950955741?check_suite_focus=true#step:6:52) we are getting from the task.

> Error: Workflows triggered by Dependabot on the "push" event run with read-only access. Uploading Code Scanning results requires write access. To use Code Scanning with Dependabot, please ensure you are using the "pull_request" event for this workflow and avoid triggering on the "push" event for Dependabot branches. See https://docs.github.com/en/code-security/secure-coding/configuring-code-scanning#scanning-on-push for more information on how to configure these events.

[Others also seem](https://github.com/cisagov/con-pca-api/commit/0d726beaa314e0229af0104fb082232eed1ef54f) to [be doing something similar](https://github.com/Skyscanner/applicationset-progressive-sync/issues/80).

### CODE REVIEW

**Comment by jamesgorrie** [comment]
@dependabot recreate

**Comment by DavidLawes** [comment]
Considering branches that aren't created by dependabot, what value do we get on running checks on push?

File: .github/workflows/snyk.yml [hunk 1]
```diff
@@ -1,14 +1,22 @@
...
+# This action runs every day at 6 AM, on every push not from dependabot, and every pull_request from dependabot
 # If the branch it's running on is main then it will run snyk monitor (reports vulnerabilities to snyk.io)
 # Otherwise it will run snyk test
 name: DCR Snyk
 
 on:
     schedule:
         - cron: '0 6 * * *'
-    push:
+    push:
```
Comment (DavidLawes) [inline .github/workflows/snyk.yml:9]: Totally unimportant: I think there's a trailing whitespace on this line?

**Comment by jamesgorrie** [comment]
@DavidLawes 
> what value do we get on running checks on push

If someone pushes to the `main` branch, or anywhere really. My thinking on this was: Keep running on push for maximum coverage, but run in `pull_request` as a workaround to give it the permissions it needs. 

Do you think that the complexity is worth keeping the maximum coverage?

File: .github/workflows/snyk.yml (hunk 1 above)
Comment (jamesgorrie) [reply .github/workflows/snyk.yml:9]:
```suggestion
    push:
```

**Review by DavidLawes — APPROVED** [review]
👍

**Pull request merged**